src/main/java/com/sonyericsson/jenkins/plugins/bfa/db/MongoDBKnowledgeBase.java

```diff
@@ -706,7 +721,7 @@ private MongoClient getMongoConnection() {
 
                     }).applyToServerSettings(builder12 -> {
             }).applyToSocketSettings(builder13 -> builder13.connectTimeout((CONNECT_TIMEOUT), TimeUnit.MILLISECONDS)).
-                    applyToSslSettings(builder14 -> builder14.enabled(tls));
+                    applyToSslSettings(builder14 -> builder14.enabled(tls)).retryWrites(disableRetryWrites);
```

Comment:
This looks so weird. Why is it inverted here and not below?

Reply:
Addressed the concern and simplified it